nextflow-genomics-cluster-ubuntu/azuredeploy.json

```diff
@@ -138,7 +138,7 @@
       "metadata": {
         "description": "[Advanced] This is best left as default unless you are an advanced user. The base URI where artifacts required by this template are located."
       },
-      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/c16cba8ead12a2dd2c3a06487e691a52f334318c"
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/875488ec32b8c7d6f08948938f8347664eb5b9d1"
```

Comment:
Artifacts need to be part of this sample...

See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

Reply:
Hi,

Thanks for the review, is this related to the SHA on the URL? I had a go at explaining this in my comment but think I mangled it a bit. 

The artifacts are still part of the sample - this URL syntax just ensures that the script is referenced at the current version. `875488ec32b8c7d6f08948938f8347664eb5b9d1` is the commit ID. I did this because we saw updates to the script breaking previously deployed clusters during testing. 

For example, without this an already deployed cluster will break, if breaking changes are made to the script, when scaling the VMSS resource. However, referencing this current commit version means a cluster deployed with the template will always use the script version present at the time of deployment. 

When the script in line 141 is updated in future the new SHA is set in the template so new deployment use this new version but old existing deployments continue unaffected.

Does this make sense? Seemed like a good idea as clusters could remain in use for a large amount of time.

Reply:
yes it does make sense - but the quick start repo is not the solution for storing artifacts for production workloads...  Everything in the samples need to be transparent.  So design this for a first time user and only that scenario.  Using the other BPs in the repo, if someone pulls this for a production workload it will work as expected.